modules/Bio/EnsEMBL/RNAProduct.pm

```diff
@@ -0,0 +1,976 @@
+  Description: Getter/setter for attribute created_date
+  Returntype : string
+  Exceptions : none
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub created_date {
+  my $self = shift;
+  $self->{'created_date'} = shift if (@_);
+  return $self->{'created_date'};
+}
+
+
+=head2 display_id
+
+  Example    : print $rnaproduct->display_id();
+  Description: This method returns a string that is considered to be
+               the 'display' identifier. For rnaproducts this is (depending on
+               availability and in this order) the stable ID, the dbID or an
+               empty string.
+  Returntype : string
+  Exceptions : none
+  Caller     : web drawing code
+  Status     : Stable
+
+=cut
+
+sub display_id {
+  my $self = shift;
+  return $self->stable_id() || $self->dbID() || '';
+}
+
+
+=head2 end
+
+  Arg [1]    : (optional) int $end - end position to set
+  Example    : $rnaproduct->end(39);
+  Description: Getter/setter for the value of end, which is a position within
+               the precursor Transcript.
+  Returntype : int
+  Exceptions : none
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub end {
+  my $self = shift;
+  $self->{'end'} = shift if (@_);
+  return $self->{'end'};
+}
+
+
+=head2 end_Exon
+
+  Arg [1]    : (optional) Bio::EnsEMBL::Exon || undef - start exon to assign
+  Example    : $rnaproduct->end_Exon($exon1);
+  Description: Getter/setter for the value of end_Exon, which denotes the
+               exon at which rnaproduct ends.
+  Returntype : Bio::EnsEMBL::Exon
+  Exceptions : thrown on wrong argument type
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub end_Exon {
+  my ($self, $exon) = @_;
+
+  if (defined($exon)) {
+
+    # Normal setter
+    assert_ref($exon, 'Bio::EnsEMBL::Exon');
+    $self->{'end_exon'} = $exon;
+
+  }
+  elsif (@_ > 1) {
+    # User has explicitly passed undef. Break connection to exon.
+    delete( $self->{'end_exon'} );
+  }
+
+  return $self->{'end_exon'};
+}
+
+
+=head2 genomic_end
+
+    Args        : None
+    Example     : $rnaproduct_genomic_end = $rnaproduct->genomic_end();
+    Description : Returns the end position of the rnaproduct in genomic
+                  coordinates on the forward strand.
+    Return type : Integer
+    Exceptions  : None
+    Caller      : General
+    Status      : Stable
+
+=cut
+
+sub genomic_end {
+  my $self = shift;
+
+  if (!exists $self->{'genomic_end'}) {
+    my $transcript = $self->transcript();
+
+    if ($transcript->strand() >= 0) {
+      $self->{'genomic_end'} =
+	$transcript->start() + ($self->end() - 1);
+    } else {
+      $self->{'genomic_end'} =
+	$transcript->end() - ($self->start() - 1);
+    }
+  }
+
+  return $self->{'genomic_end'};
+}
+
+
+=head2 genomic_start
+
+    Args        : None
+    Example     : $rnaproduct_genomic_start = $rnaproduct->genomic_start();
+    Description : Returns the start position of the rnaproduct in
```

Comment:
rnaproduct => RNA product
or should it be $rnaproduct or RNAProduct as used in previous pods?